ATB/DataAccess/Repositories/Implementations/CsvBookingRepository.cs

```diff
@@ -0,0 +1,85 @@
+using DataAccess.CsvHelperService;
+using DataAccess.Enums;
+using DataAccess.Models;
+using DataAccess.Repositories.Interfaces;
+using DataAccess.SearchCriteria;
+
+namespace DataAccess.Repositories.Implementations;
+
+public class CsvBookingRepository : IBookingRepository
+{
+    private readonly string _pathToCsv;
+    private readonly ICsvFileService<Booking> _csvFileService;
+    private List<Booking> _bookingsCache;
```

Comment:
is it practical to have a list of booking and loop over it each time I am looking for an entity Id? can you use more efficient data structure that looks up the element by O(1) time complexity?